closed/OpenJ9.gmk

```diff
@@ -256,9 +256,52 @@ build-j9 : run-preprocessors-j9
+	
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9vm29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libcuda4j29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libffi29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9thr29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libhyprtshim29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libhythr.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9dmp29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9gc29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9gc29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9gcchk29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9hookable29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9jit29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs	
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9jnichk29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9jvmti29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9prt29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9trc29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9vm29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9vmchk29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9vrb29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9zlib29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libomrsig.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libjsig.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+
+	@$(CP) -p $(OUTPUT_ROOT)/vm/j9vm_b156/libjvm.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs        
+	@$(CP) -p $(OUTPUT_ROOT)/vm/jcl/cl_se9/libjclse9_29.so $(OUTPUT_ROOT)/support/modules_libs/java.base/compressedrefs
+
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libj9shr29.so $(OUTPUT_ROOT)/support/modules_libs/openj9.sharedclasses/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libmanagement.so $(OUTPUT_ROOT)/support/modules_libs/java.management/compressedrefs
+	@$(CP) -p $(OUTPUT_ROOT)/vm/libmanagement_ext.so $(OUTPUT_ROOT)/support/modules_libs/java.management/compressedrefs
```

Comment:
From previous comments, these files are also copied into the jdk image from elsewhere. This other place should change as well, to only copy the shared libraries not included in this list.

Reply:
The files are being copied into the java.base module and not directly into the jdk image from this change. Currently the build copies the files is to do a blanket search of the vm directory for any files that matches  $(OUTPUT_ROOT)/vm/$(LIBRARY_PREFIX)*$(SHARED_LIBRARY_SUFFIX) and then copies those files into the jdk/lib/compressedrefs.

Its only after this copy has happened that the Jlink of java.base causes the files in java.base (added in this change) to "override" the versions in the jdk/lib/compressedrefs dir, although technically they are the same files as their source directory is the same.

Is it really worth coming up with an exclude list for this?

Reply:
> Is it really worth coming up with an exclude list for this?

Yes. We don't want the jdk image to work but not jlinked images. Say a new jvm library is added, it needs to be added into java.base, but if it's copied into the jdk image using $(LIBRARY_PREFIX)*$(SHARED_LIBRARY_SUFFIX) we may not realize for some time the library is missing from java.base. The best way to ensure the jdk image reflects what is created from jlink is to remove the $(LIBRARY_PREFIX)*$(SHARED_LIBRARY_SUFFIX) copy.